crates/phactory/src/system/mod.rs

```diff
@@ -1201,10 +1201,47 @@ impl<Platform: pal::Platform> System<Platform> {
                     .get_cluster_mut(&cluster_id) else {
                         return Ok(());
                     };
-                let _uploader = phala_types::messaging::AccountId(origin.clone().into());
-                let hash = cluster
-                    .upload_resource(&origin, resource_type, resource_data)
-                    .map_err(|err| anyhow!("Failed to upload code: {:?}", err))?;
+                let system_contract = cluster.system_contract().ok_or_else(|| {
+                    anyhow!(
+                        "Failed to upload resource to cluster {cluster_id:?}: No system contract"
+                    )
+                })?;
+                let result = cluster.upload_resource(&origin, resource_type, resource_data);
+                let log_handler = self.get_system_message_handler(&cluster_id);
+                if let Some(log_handler) = &log_handler {
+                    macro_rules! send_log {
```

Comment:
Looks a bit adhoc. I understand the macro is only used here to shorten the code a little bit. If there's no better way, maybe we can add some comments to highlight this `if` branch is used to deal with logs. So that we can make it easier to understand.

Reply:
The macro is inside this if scope. The other code in this scope is short enough. I think its purpose is obvious.

Reply:
I think the code is mixed with other stuff. Adding some comment to separate the lines would be better

Reply:
Added